client/src/parsec/classes.ts

```diff
@@ -0,0 +1,469 @@
+        this.device = {
+          keyFilePath: '/path',
+          organizationId: 'MyOrg',
+          deviceId: 'deviceid',
+          humanHandle: {
+            label: 'A',
+            email: 'a@b.c',
+          },
+          deviceLabel: 'a@b',
+          slug: 'slug',
+          ty: DeviceFileType.Password,
+        };
+        resolve({ok: true, value: this.device});
+      });
+    }
+  }
+}
+
+export class DeviceClaim {
+  MOCK_HOST_WAITING_TIME = 500;
+
+  handle: Handle | null;
+  canceller: Handle | null;
+  SASCodeChoices: SASCode[];
+  correctSASCode: SASCode;
+  guestSASCode: SASCode;
+  device: AvailableDevice | null;
+  greeter: HumanHandle | null;
+
+  constructor() {
+    this.handle = null;
+    this.canceller = null;
+    this.guestSASCode = '';
+    this.correctSASCode = '';
+    this.SASCodeChoices = [];
+    this.device = null;
+    this.greeter = null;
+  }
+
+  async abort(): Promise<void> {
+    if (this.canceller !== null && window.isDesktop()) {
+      libparsec.cancel(this.canceller);
+    }
+    if (this.handle !== null && window.isDesktop()) {
+      libparsec.claimerGreeterAbortOperation(this.handle);
+    }
+    this.canceller = null;
+    this.handle = null;
+    this.guestSASCode = '';
+    this.correctSASCode = '';
+    this.SASCodeChoices = [];
+    this.device = null;
+    this.greeter = null;
+  }
+
+  _assertState(nullCanceller: boolean, nullHandle: boolean): void {
+    if (nullCanceller && this.canceller !== null) {
+      throw Error('Canceller should be null');
+    } else if (!nullCanceller && this.canceller === null) {
+      throw Error('Canceller should not be null');
+    }
+    if (nullHandle && this.handle !== null) {
+      throw Error('Handle should be null');
+    } else if (!nullHandle && this.handle === null) {
+      throw Error('Handle should not be null');
+    }
+  }
+
+  async retrieveInfo(invitationLink: string):
+  Promise<Result<UserOrDeviceClaimInitialInfoUser, ClaimerRetrieveInfoError>> {
+    function eventCallback(event: ClientEvent): void {
+      console.log('On event', event);
+    }
+
+    this._assertState(true, true);
+
+    if (window.isDesktop()) {
+      const clientConfig = getClientConfig();
+      const result = await libparsec.claimerRetrieveInfo(clientConfig, eventCallback, invitationLink);
+      if (result.ok) {
+        this.handle = result.value.handle;
+        this.greeter = result.value.greeterHumanHandle;
+      }
+      return result as Result<UserOrDeviceClaimInitialInfoUser, ClaimerRetrieveInfoError>;
+    } else {
+      await _wait(this.MOCK_HOST_WAITING_TIME);
+      this.handle = DEFAULT_HANDLE;
+      this.greeter = {
+        email: 'gale@waterdeep.faerun',
+        // cspell:disable-next-line
+        label: 'Gale Dekarios',
+      };
+      return new Promise<Result<UserOrDeviceClaimInitialInfoUser, ClaimerRetrieveInfoError>>((resolve, _reject) => {
+        resolve({ok: true, value: {
```

Comment:
This returns a `UserOrDeviceClaimInitialInfoDevice`, tag should be `Device`, no need for any `claimerEmail`.